adding exercises (#2)
r4ds/bookclub-deepr: 01_introduction.Rmd

**Learning objectives:**

> - What this book is about! 

Reviewer: ```suggestion
- Describe what this book is about! 
```

In general, aim for learning objectives to complete the sentence "After you read this chapter, you should be able to..."

> - Run R in **interactive**, **batch mode** and **semi-interactive**

Reviewer: ```suggestion
- Run R in **interactive mode**, **batch mode** and **semi-interactive mode**
```

> - Create, import and work with one of the basic data type: **numeric vector**

Reviewer: ```suggestion
- Create, import and work with one of the basic data types: **numeric vector**
```

## Preface 1 {-}

- R is a language and an environment

  * Graphics capabilities. 
  * Help system. 

>   * Interfacing compile code. 

Reviewer: ```suggestion
  * Interface to compiled code. 
```

  * A package systenm + package repositories (CRAN) 
  * A 👌 lively community!
  
## Preface 2 {-}

The author does not like: 

- the way R is taught in a stats course 

- the tidyverse

> He advocates for an approach that could help us solving every data intense problem and acquire skills relevant to other language. 

Reviewer: ```suggestion
He advocates for an approach that could help us solve every data intense problem and acquire skills relevant to other languages. 
```

## Preface 3 {-}

> The book follow more more data types approach than a workflow approach. 

Reviewer: ```suggestion
The book follows more of a data-types approach than a workflow approach. 
```

1. Basic types: 
    * Atomic: logical (TRUE/FALSE), numeric, character. 
    * Function. 
    * List. 

2. Compound types: 
    * Factor. 
    * Matrix. 
    * data.frame.  
    * formula. 
    
3. Advanced material: 
  * Symbol, call, expressions: metaprograming. 
  * Environment: scoping / store/access objects   
  * Interface compiled code

## Introduction 1. {-}

### Interactive mode

Go into R and use `cat()` ! 

`>` see the prompt 

> `+` as for more

Reviewer: ```suggestion
`+` is for more
```

> and we can use `ESC` or `CTRL+C` to abord input! 

Reviewer: ```suggestion
and we can use `ESC` or `CTRL+C` to abort input! 
```
